File: 509.fibonacci-number.js
/*
 * @lc app=leetcode id=509 lang=javascript
 *
 * [509] Fibonacci Number
 */

/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
	if (n === 0) return 0;
	if (n === 1) return 1;
	let f = [ 0, 1 ];
	let i = 2;
	while (i <= n) {
		f.push(f[i - 1] + f[i - 2]);
		i++;
	}
	return f[n];
};
fib(2);

File: java/509.fibonacci-number.java
/*
 * @lc app=leetcode id=509 lang=java
 *
 * [509] Fibonacci Number
 */

// @lc code=start
class Solution {
    public int fib(int N) {

        int[] f = new int[N + 1];
        if (N == 0)
            return 0;
        f[0] = 0;
        f[1] = 1;
        for (int i = 2; i <= N; i++) {
            f[i] = f[i - 1] + f[i - 2];
        }
        return f[N];
    }
}
// @lc code=end
